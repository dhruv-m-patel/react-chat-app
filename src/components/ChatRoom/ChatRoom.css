.chatroom {
  height: 90vh;
  width: 100%;
  color: var(--black);
}
