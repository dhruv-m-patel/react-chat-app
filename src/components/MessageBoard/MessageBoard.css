@import '../../styles/variables.css';

.messageBoard {
  padding: 0.5rem 1rem;
  height: 92%;
}

.dark.messageBoard {
  background-color: var(--backgroundColor);
  color: var(--white);
}

.light.messageBoard {
  background-color: var(--grey);
}
