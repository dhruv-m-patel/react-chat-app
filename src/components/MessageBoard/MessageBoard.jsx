import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames/bind';
import styles from './MessageBoard.css';
import PageContext from '../../context/PageContext';

const cx = classnames.bind(styles);

export default function MessageBoard({
  messages,
}) {
  const pageContext = useContext(PageContext);

  return (
    <div className={cx(pageContext.theme, 'messageBoard')}>
      <ul className={cx('messageList')}>
        {messages.map(message => message && (
          <li key={message.parts[0].payload.content} className={cx('message', message.senderId)}>
            <div className={cx('sender')}>
              {message.senderId}
            </div>
            <div className={cx('text')}>
              {message.parts[0].payload.content}
            </div>
          </li>
        ))}
      </ul>
    </div>
  );
}

MessageBoard.propTypes = {
  messages: PropTypes.arrayOf(
    PropTypes.shape({
      senderId: PropTypes.string,
      parts: PropTypes.arrayOf(
        PropTypes.shape({
          payload: PropTypes.shape({
            content: PropTypes.string,
          }),
        }),
      ).isRequired,
    }),
  ),
};

MessageBoard.defaultProps = {
  messages: [],
};
