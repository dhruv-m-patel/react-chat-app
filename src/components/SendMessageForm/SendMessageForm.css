.sendMessageForm {
  padding: 0.25rem;
  height: 8%;
  border-top: 1px solid var(--blue);

  & input {
    height: 30px;
    font-size: 0.8rem;
  }

  & button {
    height: 30px;
    font-size: 0.8rem;
  }
}

.dark.sendMessageForm {
  background-color: var(--backgroundColor);
  color: var(--white);
}

.light.sendMessageForm {
  background-color: var(--grey);
}
