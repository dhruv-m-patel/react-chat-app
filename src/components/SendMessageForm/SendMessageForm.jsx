import React, { useContext } from 'react';
// import PropTypes from 'prop-types';
import classnames from 'classnames/bind';
import styles from './SendMessageForm.css';
import PageContext from '../../context/PageContext';

const cx = classnames.bind(styles);

export default function SendMessageForm() {
  const pageContext = useContext(PageContext);

  return (
    <div className={cx(pageContext.theme, 'sendMessageForm')}>Send Message Form</div>
  );
}

// SendMessageForm.propTypes = {

// };

// SendMessageForm.defaultProps = {

// };
