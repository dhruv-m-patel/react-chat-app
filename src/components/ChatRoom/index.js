import { connect } from 'react-redux';
import ChatRoom from './ChatRoom';

function mapStateToProps({ config }) {
  return {
    instanceLocator: config.instanceLocator,
    receiver: config.receiverUser,
    instanceUrl: config.authToken,
  };
}

function mapDispatchToProps(dispatch) {
  return {};
}

export default connect(mapStateToProps, mapDispatchToProps)(ChatRoom);
