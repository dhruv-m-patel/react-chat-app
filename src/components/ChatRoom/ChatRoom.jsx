import React from 'react';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Container from 'react-bootstrap/Container';
import classnames from 'classnames/bind';
import Page from '../Page';
import styles from './ChatRoom.css';
import MessageBoard from '../MessageBoard';
import SendMessageForm from '../SendMessageForm';
import Smartphone from '../Smartphone';

const cx = classnames.bind(styles);

export default function ChatRoom() {
  return (
    <Page>
      <Row>
        <Col xs={{ offset: 1, span: 10 }} sm={{ offset: 4, span: 4 }}>
          <Container className={cx('chatroom')}>
            <Smartphone>
              <MessageBoard />
              <SendMessageForm />
            </Smartphone>
          </Container>
        </Col>
      </Row>
    </Page>
  );
}
