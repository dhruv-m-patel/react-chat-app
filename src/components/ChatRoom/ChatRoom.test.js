import Tester from '../../../tests/Tester';
import ChatRoom from './ChatRoom';
import MessageBoard from '../MessageBoard';
import SendMessageForm from '../SendMessageForm'

const tester = new Tester();

describe('ChatRoom', () => {
  test('it should render', () => {
    const { component } = tester.getShallowInstance(
      ChatRoom,
      {
        instanceLocator: 'value',
        receiver: 'receiver',
        instanceUrl: 'url',
      },
    );

    expect(component).toBeDefined();
    expect(component.find(MessageBoard).length).toEqual(1);
    expect(component.find(SendMessageForm).length).toEqual(1);
  });
});
