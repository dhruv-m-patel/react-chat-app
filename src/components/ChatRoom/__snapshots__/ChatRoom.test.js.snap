// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatRoom it should render 1`] = `
<div
  className="Page__page___11SaW Page__lightTheme___ZzJ_L container-fluid"
>
  <div
    className="Page__navBar___3VWqc"
  >
    <div
      className="row"
    >
      <div
        className="col-10"
      >
        <div
          className="Page__app___25fET"
        >
          <header
            className="Page__appHeader___3GNlT"
          >
            <img
              alt="logo"
              className="Page__appLogo___3bWCs"
              src="/images/react.svg"
            />
            React Chat App
          </header>
        </div>
      </div>
      <div
        className="col-2"
      >
        <div
          className="Page__textRight___80Bv0"
        >
          Dark Mode
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-toggle-off fa-w-18 fa-2x Page__clickable___17a83 Page__padTop10px___2MDcB"
            data-icon="toggle-off"
            data-prefix="fas"
            focusable="false"
            onClick={[Function]}
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M384 64H192C85.961 64 0 149.961 0 256s85.961 192 192 192h192c106.039 0 192-85.961 192-192S490.039 64 384 64zM64 256c0-70.741 57.249-128 128-128 70.741 0 128 57.249 128 128 0 70.741-57.249 128-128 128-70.741 0-128-57.249-128-128zm320 128h-48.905c65.217-72.858 65.236-183.12 0-256H384c70.741 0 128 57.249 128 128 0 70.74-57.249 128-128 128z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <br />
  <div
    className="Page__content___rCsmc container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-4 col-10 offset-sm-4 offset-1"
      >
        <div
          className="ChatRoom__chatroom___1rhtu container"
        >
          <div
            className="light Smartphone__smartphone___VmVC5"
          >
            <div
              className="Smartphone__content___12Svw"
            >
              <div
                className="MessageBoard__light___3wjYC MessageBoard__messageBoard___12gbl"
              >
                Message Board
              </div>
              <div
                className="SendMessageForm__light___3WIoQ SendMessageForm__sendMessageForm___2nHAp"
              >
                Send Message Form
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
